### ISSUE TITLE
Problem with set_rate_BC when it gets array of pores and arrays of rate values

### ISSUE THREAD

**Issue by Ni2M** [PMEAL/OpenPNM#1479]
Using an array of values to set a rate BC for an array of pores (the same length) in ionic conductantion algorithm. For some reason it gives an error that goes back to generic transport _set_BC self['pore.bc_' + bctype][intersect] += values line. With error "operands could not be broadcast together with shapes (0,) (1808,) (0,)" .

**Comment by Ni2M** [comment]
Hey @ma-sadeghi , I provided a simple example that helps us to address the issue. I appreciate your time and consideration if you could please review the following example. There is still the problem with set_rate_BC, while set_value_BC works fine.
```python
import openpnm as op
import numpy as np
import matplotlib.pyplot as plt
net = op.network.Cubic(shape=[100, 100, 1])
geom = op.geometry.StickAndBall(network=net)
water = op.phases.Water(network=net, name="water")
#net has 10000 pores
#we chose left+right pores (200 pores) as borders with constant value of potential
#we use the rest of the pores (9800) as rate BC (given their rate values as random values)
ps_border=net.pores(["left","right"], mode="or")
ps_internal=np.setdiff1d(net.Ps, ps_border)
BC1=np.ones(np.size(ps_border))*0.02
BC2=np.random.rand(len(ps_internal))
#BC2=np.ones(np.size(ps_internal))*0.1

#%%this cell gives an error (when we have set_rate_BC)
ic = op.algorithms.IonicConduction(network=net, phase=water)
water["throat.ionic_conductance"]=33.5/net["throat.length"] 
ic.set_value_BC(pores=ps_border, values=BC1)
ic.set_rate_BC(pores=ps_internal, values=BC2)
ic.settings['rxn_tolerance'] = 1e-12
ic.run()
water.update(ic.results())

#%%#this cell works properly (when we change set_rate to set_value)
ic = op.algorithms.IonicConduction(network=net, phase=water)
ic.set_value_BC(pores=ps_border, values=BC1)
ic.set_value_BC(pores=ps_internal, values=BC2)
ic.settings['rxn_tolerance'] = 1e-12
ic.run()
water.update(ic.results())
pot = water["pore.potential"]
pot2 = np.rot90(pot.reshape(net._shape).squeeze())
plt.imshow(pot2)
plt.colorbar()
#%%`
```

**Comment by ma-sadeghi** [comment]
This is now addressed in #1489
